bigtable-hbase-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -819,93 +832,174 @@ public boolean isTableAvailable(TableName tableName, byte[][] splitKeys) throws
   @Override
   public void snapshot(String snapshotName, TableName tableName)
       throws IOException, SnapshotCreationException, IllegalArgumentException {
-    throw new UnsupportedOperationException("snapshot");  // TODO
+    waitForOperation(snapshotTable(snapshotName, tableName));
+  }
+
+  private Operation snapshotTable(String snapshotName, TableName tableName)
+      throws IOException {
+    return bigtableTableAdminClient.snapshotTable(SnapshotTableRequest.newBuilder()
```

Comment:
Sorry! I don't know if you have addressed my previous comment here. Do we really want default to Max TTL? This makes me think of having a better default on server side.